internal/service/iam/role_policies_exclusive_test.go

```diff
@@ -0,0 +1,330 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package iam_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/service/iam/types"
+	sdkacctest "github.com/hashicorp/terraform-plugin-testing/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/hashicorp/terraform-provider-aws/internal/acctest"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/errs"
+	tfiam "github.com/hashicorp/terraform-provider-aws/internal/service/iam"
+	"github.com/hashicorp/terraform-provider-aws/names"
+)
+
+func TestAccIAMRolePoliciesExclusive_basic(t *testing.T) {
```

Comment:
There are exclusive functionality tests in IAM role that should be included here to make sure all the various possibilities (out of band additions, deletions, etc.) also work with the new resource. Approximately some version of these tests: https://github.com/hashicorp/terraform-provider-aws/blob/18b873b0c985481efbfface3ec63654a1f4c13a6/internal/service/iam/role_test.go#L591-L1044

Reply:
Added tests for out of band inline policy additions and deletions. The empty `inline_policy` block case is covered by an existing test, and some of the others around the behavior of the block when unset won't apply to the standalone resource. I think this should now cover everything the existing role test suite does.